src/Orleans.Runtime/Catalog/ActivationCollector.cs

```diff
@@ -513,7 +513,7 @@ private async Task DeactivateActivationsFromCollector(List<ICollectibleGrainCont
             logger.LogInformation((int)ErrorCode.Catalog_ShutdownActivations_1, "DeactivateActivationsFromCollector: total {Count} to promptly Destroy.", list.Count);
             CatalogInstruments.ActiviationShutdownViaCollection();
 
-            Action<Task> signalCompletion = task => mtcs.SetOneResult();
+            void signalCompletion(Task task) => mtcs.SetOneResult();
```

Comment:
This one should be reverted - it's going to cause potentially multiple delegate allocations (one for each list item, instead of one overall).